executor/executor_test.go

```diff
@@ -628,6 +628,70 @@ func (s *testSuite) TestLoadData(c *C) {
 	}
 }
 
+func (s *testSuite) TestLoadDataEscape(c *C) {
+	defer testleak.AfterTest(c)()
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test; drop table if exists load_data_test;")
+	tk.MustExec("CREATE TABLE load_data_test (id INT NOT NULL PRIMARY KEY, value TEXT NOT NULL) CHARACTER SET utf8")
+	_, err := tk.Exec("load data local infile '/tmp/nonexistence.csv' into table load_data_test")
+	c.Assert(err, IsNil)
+
+	ctx := mock.NewContext()
+	ctx.Store = s.store
+	variable.BindSessionVars(ctx)
+	domain, err := domain.NewDomain(s.store, 1*time.Second)
+	c.Assert(err, IsNil)
+	is := domain.InfoSchema()
+	c.Assert(is, NotNil)
+	tbl, err := is.TableByName(model.NewCIStr("test"), model.NewCIStr("load_data_test"))
+	c.Assert(err, IsNil)
+	fields := &ast.FieldsClause{Terminated: "\t"}
+	lines := &ast.LinesClause{Starting: "", Terminated: "\n"}
+	ld := executor.NewLoadDataInfo(make([]types.Datum, 2), ctx, tbl)
+	ld.FieldsInfo = fields
+	ld.LinesInfo = lines
+
+	deleteSQL := "delete from load_data_test"
+	selectSQL := "select * from load_data_test;"
+	type testCase struct {
+		data1    []byte
+		data2    []byte
+		expected []string
+		restData []byte
+	}
+	// data1 = nil, data2 = nil, fields and lines is default
+	_, err = ld.InsertData(nil, nil)
+	c.Assert(err, IsNil)
+	r := tk.MustQuery(selectSQL)
+	r.Check(nil)
```

Comment:
The table struct is not the same, so it's hard to reuse all the code or put them in one function.
I'll try to extract some common function.